src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -148,6 +148,13 @@ public abstract partial class JsonNamingPolicy
         public bool NameEquals(string text) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial struct JsonDocumentOptions
+    {
+        private int _dummyPrimitive;
+        public bool AllowTrailingCommas { readonly get { throw null; } set { } }
+        public System.Text.Json.JsonCommentHandling CommentHandling { readonly get { throw null; } set { } }
+        public int MaxDepth { get { throw null; } set { } }
```

Comment:
Why are AlllowTrailingCommas and CommentHandling readonly, but MaxDepth isn't?

Reply:
The "readonly" was added as part of the auto-generation of the ref. It looks like there is a bug where it isn't marking all the property getters as readonly (specifically for an integer property).

I don't want to manually change this and keep fighting the auto-gen. Would it be considered a breaking change to mark a property getter readonly in the future?

@tannergooding, @safern, @ericstj - do you know why the `MaxDepth` property is not getting marked as readonly here?

If I regenerate the ref from the source directory, the `readonly` keyword on the `CommentHandling` getter gets removed too (only AllowTrailingCommas retains it). Does the `readonly` keyword only apply to auto-properties?

Reply:
Looking back at https://github.com/dotnet/arcade/issues/3010#issuecomment-500297307

> Auto properties on mutable structs have their getter marked readonly starting with C# 8. 

The `AllowTrailingCommas` is the only auto-prop. The other two are no longer auto-props.

Reply:
Right, `readonly` is added to auto property getters by default.

It would probably be good to mark most of our non-auto property getters as readonly as well; since they shouldn't be mutating state. Note: it was also already determined that using unsafe code to allow lazy initialization in a `readonly` member should be fine as well (if that comes up as a concern).